lib/dom_parser.rb

```diff
@@ -0,0 +1,93 @@
+require_relative 'file_loader.rb'
+
+Tag = Struct.new(:text, :type, :classes, :id, :name, :children, :parent, :closing, :depth)
+
+
+class DomParser
+
+  TYPE_REGEX = /<[a-z]*[0-9]*/
+  CLASS_REGEX = /class=('|")[[a-z0-9]*\W*\s]*?('|")/
+  ID_REGEX = /id=('|")[[a-z0-9]*\W*]*?('|")/
+  NAME_REGEX = /name=('|")[[a-z0-9]*\W*]*?('|")/
+  TEXT_REGEX = />(\s*.*?\s*)</
+
+  ALL_TAG_REGEX = /(<[a-z]*.*?>)/
+  OPEN_TAG_REGEX = /<[a-z].*?>/
+  # SELF_CLOSING_REGEX = /^(.*?>.*?)>/
+  # ROOT_CLOSING = /<\/html>/
+
+
+  attr_reader :root, :tag_list, :tag_structs, :html_string, :tag_text, :tree
+
+  def initialize
+    @file_loader = FileLoader.new
+    @html_string = ""
```

Comment:
It looks like this instance variable is only used in the (find all tags) method. You can just pass it as a parameter from the build_tree method to find_all_tags instead of declaring it an instance variable.